### DISCUSSION TITLE
What Should I learn first?

### DISCUSSION THREAD

**Discussion by aditya-coding** [community/community#29780] [Codespaces] [answered]
Hi I am Aditya Singh from 8th STD. AND I AM FULL DEDICATED AND PASSIONATE ABOUT CODING.. 

BUT I'M BIT CONFUSED WHAT SHOULD I LEARN FIRST PYTHON OR C/C++/C# CAN SOME OF THE DEVELOPERS CAN ADVISE ME.. 

MY PRIORITY IS TO LEARN LANGUAGES AND TO CREATE LOT IF THINGS AT AN EARLY AGE.... PLEASE HELP ME..

**Comment by airtower-luna** [comment]
First learn to turn off capslock. Writing in all-caps is rude, like yelling into people's faces.

For a first programming language to learn Python is an excellent choice. :snake: It has gentle learning curve: You can make small useful things pretty quickly, and more complex things as you learn more. And you can apply it in almost every field later, if you want.

**Comment by jfullstackdev** [comment, marked as answer]
let me just share my own experience, I was not even a graduate of any IT or CS fields ( I was a basic Educ grad ) but it's now my job, one as an IT / CS Instructor and two, as a Full Stack Developer of PHP / Laravel

I started with formulas in Excel, was able to have a flat-file database for the Sportsfests of the school, then proceeded learning PureBasic as I was too inspired although at first that was very discouraging, as in I could not even understand what are variables and why we need them, I started learning in a very slow manner, but one thing is for sure, you need to build even small programs to practice what you are learning, in any programming language

then next is Java, again, that was very discouraging because of the syntax unlike PureBasic, but I still managed to learn it and build programs using it, then learning its framework Spring Boot

the same thing happened when I learned PHP / Laravel

so going back, as for the starter, focus maybe on Python and JavaScript, the first time I encountered Python, I've learned the basics within 5 minutes because I have prior knowledge already

in the end, I realized there are core concepts no matter what language you are using, because a programming language is just a tool, an IT company can even changed the technologies they are using but you still need to adapt to it while working 

https://github.com/jdevstatic/programming-core-concepts

and lastly, your determination and dedication to it, now matter what endeavor you are taking

**Reply by aditya-coding** [reply]
Thank You SO MUCH For your Amazing ANSWER MY VERDICT IS PYTHON... **THANK YOU SO MUCH**

**Comment by sayhiben** [comment]
First, learn that this discussion board is for discussing GitHub Codespaces, and not your career

**Reply by aditya-coding** [reply]
Just SHUT UP!

**Comment by ImaniAN** [comment]
Hi @aditya-coding I just took a look at your GitHub profile and its like 2 days old lol. That not an issue but it explains why you asked a question on the wrong board and that's okay, its expected that you will make mistakes and stumble along your journey. Just do it with respect and grace. 

**Now to respond to your question.** 

8th STD (lol)... Doesn't really tell us much about your capacity but I will make some assumption in order to provide some useful feedback. 

Standard 8 where I am from is between grade 9 and 11 (High School or Secondary School) depending on the school you go to and the curriculum. 

Because you might not be done with high school. I would advise you to spend between 60%-70% of you learning time. familiarizing yourself with Math concepts that are used in programming.(You will benefit greatly from have having a novice understanding of ALL the topics) 

_Here are some topics to give you an idea:_
- **Number Systems**
- - Natural Numbers
Integers
Irrational/ Rational Numbers
Nonreal/Real
Complex
- **Structures 1**
- - Algebra
Equations
Linear Algebra
Matrices
Vectors
- **Spaces**
- - Geometry
Trigonometry
- **Changes**
- - Calculus
Differential Equations
Vector Calculus
Chaos Theory
- **Chances**
- - Probability
Statistics
- **Structures 2**
- - Measure Theory
Number Theory
Order Theory
Category  Theory
Group  Theory
Partition  Theory
Graph Theory
Discrete Theory
- **Topology**
- - Differential Geometry
- **Numerical Analysis**
- - Intro
- **Computational Theory**
- - Intro

When you feel like you have a reasonable understanding of some of these topics, sit down and decide what operating system you are going to use to learn programming. (Suggestion: Linux, Windows And MacOS).

Once you have selected and version or distribution of your operating system.

Read about the [OSI model](https://en.wikipedia.org/wiki/OSI_model) , This will give you an understanding of how the internet works, which is super important just incase you run into any issue in life.

You already have a GitHub account so here is Gold Star for your efforts⭐✨ Well done!!! You are making the right moves! 

**Download** [_VSCode_](https://code.visualstudio.com/download). A text editor to help you write code

**Download** [_Git Version Control_](https://git-scm.com/downloads) A version control system. (You might want to familiarize yourself with version control systems, you will use them a lot... like a lot lot.) [A link I use when I need to be remember certain version control concepts](https://betterexplained.com/articles/a-visual-guide-to-version-control/)

Once you have gotten to this stage you are now ready to be a programmer. 
Start with HTML, CSS and JavaScript. Use this website to get you started. [_W3Schools_](https://www.w3schools.com/html/default.asp). 

**W3cSchools will provide examples, exercises and projects FOR FREE!!!**

After learning HTML, CSS and JavaScript. You will be ready to take on a "REAL" programming language and start doing "SERIOUS" programming.

At this level of you programming journey. I recommend:
0. JavaScript (**_Commit to it, its worth it_**)
1. Python (Short learning curve, many use cases)
2. C# (Microsoft Certifications are extremely valuable when passed)
3. SQL (MySQL, PostgreSQL and T-SQL)
4. Java (Learn at your own risk... Very well paying jobs with enough experience) 
5. PHP (Learn at your own risk... Very well paying jobs with enough experience)

If you don't like any of these languages, which is normal, take a look at RUBY, GO and RUST (popular programming languages with good communities lots of developer support). 

If after all this, you still don't like any of the programming languages. You can still continue learning more about ICT and Software Programming by getting a Cloud Certification from AZURE, Google Cloud, AWS Cloud or CISCO. (The Cloud Certifications are my personal recommendations, feel free to pursue what you want/like)

Software Programming is mostly about practice, so make it a habit to create GitHub repo's to have portfolio of evidence so that you can showcase you competencies, skills, knowledge and expertise. 

If you get lost or confused, please feel free to reference this document which is a Microsoft Project file which outlines all the things I have mentioned. [Full Stack Developer RoadMap](https://github.com/ImaniAN/A-FullStack-Dev-RoadMap)

You can also reach me at [Contact Information](https://github.com/ImaniAN/ImaniAN#how-to-reach-me)  

I hope this helps you get an understanding of what is needed to be a good programmer in terms of learning. 

I hope things work out well for you and in future, don't just post questions anywhere or wherever, lol. More attention will be given to your mistake than to actually helping solve your problem...
